task_processing/plugins/mesos/execution_framework.py

```diff
@@ -178,19 +181,22 @@ def unblacklist_slave(self, agent_id, timeout):
             self.blacklisted_slaves = \
                 self.blacklisted_slaves.remove(agent_id)
 
-    def enqueue_task(self, task_config):
+    def enqueue_task(self, task_config, task_id):
+        mesos_task_id = ':'.join(task_id)
```

Comment:
mesos_task_id is the actual value that goes to mesos over a wire. task_id is what lives in our stack. execution_framework handles mapping one to another and making sure neither leaks where it doens't belong.